ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -325,6 +326,42 @@ applyPatchHeapClosures _ = do
             "reallyUnsafePtrEqualityUpToTag# :: Any -> Any -> Int#\nreallyUnsafePtrEqualityUpToTag# _ _ = 0#\n"
         =<< readFile' file
 
+-- Workaround lack of newer ghc-prim 12/3/2019
+-- (https://gitlab.haskell.org/ghc/ghc/commit/705a16df02411ec2445c9a254396a93cabe559ef)
+applyPatchGhcPrim :: GhcFlavor -> IO ()
+applyPatchGhcPrim ghcFlavor = do
+    let tysPrim = "compiler/prelude/TysPrim.hs"
+    when (ghcFlavor == GhcMaster) (
+      writeFile tysPrim .
+          replace
+              "bcoPrimTyConName              = mkPrimTc (fsLit \"BCO\") bcoPrimTyConKey bcoPrimTyCon"
+              "bcoPrimTyConName              = mkPrimTc (fsLit \"BCO#\") bcoPrimTyConKey bcoPrimTyCon"
+        =<< readFile' tysPrim
```

Comment:
Does the [`LiftedRep --> UnliftedRep`](https://gitlab.haskell.org/ghc/ghc/commit/705a16df02411ec2445c9a254396a93cabe559ef#0188aac58d5dfb76292827933bb504a964ed7c01_1058_1061) also need to be patched?

Reply:
Wow. Well spotted. Yes it does and I'm at a loss to explain how that got through without it! Fixing and testing it now. Thank-you!